src/lib.rs

```diff
@@ -53,3 +55,10 @@ impl Behaviour {
         }
     }
 }
+
+pub fn load_secret_key_from_file(path: impl AsRef<Path>) -> Result<SecretKey> {
+    let contents = fs::read_to_string(path)?;
```

Comment:
`String`s in Rust enforce UTF8. I doubt the bytes of secret keys form valid UTF8 all the time :)

Best to read the content as bytes right away.

Also, best to use tokio's async APIs here. Shouldn't matter much but it is good to develop a habit of not mixing blocking std APIs with async code.